sdk/internal/mock/mock.go

```diff
@@ -50,14 +50,27 @@ func NewServer() (*Server, func()) {
 	return s, func() { s.srv.Close() }
 }
 
-// NewTLSServer creates a new Server object.
+// NewTLSServer creates a new Server object and applies any additional configurations
+// provided on the httptest.Server.
 // The returned close func must be called when the server is no longer needed.
-func NewTLSServer() (*Server, func()) {
+// It will return nil for both the server and close func if it encountered an error
+// when configuring HTTP2 for the new TLS server.
+func NewTLSServer(config ...func(*httptest.Server)) (*Server, func()) {
```

Comment:
This is a great start, but limits us to configuring `*httptest.Server` only.  Take a look at `ResponseOption`, if we apply that same pattern it provides more flexibility.

While you're here, could you please apply the same variadic configuration change to `NewServer()` for consistency.

Reply:
Will add the variadic config to NewServer as well. 

As to the ResponseOption config, are you talking about providing functions that provide defaults and return an interface with a private method? I had wanted to have something that allows us to modify the Mock package server without providing a common default...like this implementation for http2 servers. So I could think of adding an interface with a public method that could then provide us with even more flexibility. Is that better? Or do you think this should be exactly the same as ResponseOption?

Reply:
They wouldn't have to provide defaults, just access to the things they configure.  Something like this.
```go
func WithTLSConfig(cfg *tls.Config) ServerOption {
	return serverOption(func(s *Server) {
		s.srv.TLS = cfg
	})
}
```

Reply:
Ah gotcha. Will finish implementing like that.

Reply:
Done